mmdet/models/data_preprocessors/track_data_preprocessor.py

```diff
@@ -0,0 +1,251 @@
+                self.pad_gt_masks(data_samples)
+
+        if training and self.batch_augments is not None:
+            for batch_aug in self.batch_augments:
+                # We only support T==1 when using batch augments.
+                # Only yolox need batch_aug, and yolox can only process
+                # (N, C, H, W) shape.
+                # The shape of `inputs` is (N, T, C, H, W), hence, we use
+                # inputs[:, 0] to change the shape to (N, C, H, W).
+                assert inputs.size(1) == 1 and len(
+                    data_samples[0]
+                ) == 1, 'Only support the number of sequence images equals to 1 when using batch augment.'  # noqa: E501
+                det_data_samples = [
+                    track_data_sample[0] for track_data_sample in data_samples
+                ]
+                aug_inputs, aug_det_samples = batch_aug(
+                    inputs[:, 0], det_data_samples)
+                inputs = aug_inputs.unsqueeze(1)
+                for track_data_sample, det_sample in zip(
+                        data_samples, aug_det_samples):
+                    track_data_sample.video_data_samples = [det_sample]
+
+        return dict(inputs=inputs, data_samples=data_samples)
+
+    def _get_pad_shape(self, data: dict) -> Dict[str, List]:
+        """Get the pad_shape of each image based on data and pad_size_divisor.
+
+        Args:
+            data (dict): Data sampled from dataloader.
+
+        Returns:
+            Dict[str, List]: The shape of padding.
+        """
+        batch_pad_shape = dict()
+        batch_pad_shape = []
+        for imgs in data['inputs']:
+            # The sequence images in one sample among a batch have the same
+            # original shape
+            pad_h = int(np.ceil(imgs.shape[-2] /
+                                self.pad_size_divisor)) * self.pad_size_divisor
+            pad_w = int(np.ceil(imgs.shape[-1] /
+                                self.pad_size_divisor)) * self.pad_size_divisor
+            pad_shapes = [(pad_h, pad_w)] * imgs.size(0)
+            batch_pad_shape.append(pad_shapes)
+        return batch_pad_shape
+
+    def pad_gt_masks(self, data_samples: Sequence[TrackDataSample]) -> None:
+        """Pad gt_masks to shape of batch_input_shape."""
+        if 'masks' in data_samples[0][0].get('gt_instances', None):
+            for track_data_sample in data_samples:
+                for i in range(len(track_data_sample)):
+                    det_data_sample = track_data_sample[i]
+                    masks = det_data_sample.gt_instances.masks
+                    # TODO: whether to use BitmapMasks
+                    assert isinstance(masks, BitmapMasks)
+                    batch_input_shape = det_data_sample.batch_input_shape
+                    det_data_sample.gt_instances.masks = masks.pad(
+                        batch_input_shape, pad_val=self.mask_pad_value)
+
+
+def stack_batch(tensors: List[torch.Tensor],
```

Comment:
Is it possible to simply extend mmengine's s-function to support it? If so, it is recommended to add a TODO note for subsequent integration into mmengine.